### ISSUE TITLE
Loss function - unpacked variables

### ISSUE THREAD

**Issue by afoltyn** [rimchang/Faster-RCNN-Pytorch-Simple#2]
Hey,
this project was really helpful to understand Faster R-CNN. So thanks :). But I think I found a small error that could contribute  to lower performance.
In the loss calculation the pytorch-Variable` frcnn_bbox_pred ` is unpacked and multiplied and after that casted to a Variable again:
` frcnn_bbox_pred = to_var(torch.mul(frcnn_bbox_pred.data, frcnn_bbox_inside_weights))`
To due this step` .backward()` can not go further than this point. If you visualize the graph you will see it. 
The following should work: 
```
rcnn_bbox_inside_weights = torch.autograd.Variable(rcnn_bbox_inside_weights, requires_grad=False)
frcnn_bbox_pred = torch.mul(frcnn_bbox_pred, frcnn_bbox_inside_weights)
```
The same step is in the rpn_loss calculation.

**Comment by rimchang** [comment]
wow!!! very helpful.
I will check it soon.
